### ISSUE TITLE
Workflow completes successfully, but site URL returns a 404

### ISSUE THREAD

**Issue by ianbayne** [andrewmcodes/bridgetown-gh-pages-action#4]
# Bug Report

## Describe the bug

Hi there! Thanks for the GH actions script! I'm trying to make use of it to deploy my Bridgetown site to GH Pages, but even though the workflow completes successfully, when I navigate to my site's URL, the server is returning a 404.

# Reproduction Steps

1. Create a new Bridgetown site.
2. Create a deploy workflow with content that's the same as either the Basic, Manual, or Advanced usages (edited of course as necessary) on https://github.com/andrewmcodes/bridgetown-gh-pages-action.
3. Commit and push.
4. Wait for GH Action to finish.
5. Navigate to site's URL.
6. See 404 page.

# Expected behavior

Site should build and deploy, and the server should return it successfully.

## Stacktrace

<img width="871" alt="Screen Shot 2021-04-17 at 13 03 03" src="https://user-images.githubusercontent.com/10363087/115101248-68a05b00-9f7d-11eb-9090-02184b300969.png">
<img width="874" alt="Screen Shot 2021-04-17 at 13 03 16" src="https://user-images.githubusercontent.com/10363087/115101258-6e963c00-9f7d-11eb-96fe-0f3794c2cb35.png">
<img width="881" alt="Screen Shot 2021-04-17 at 13 03 29" src="https://user-images.githubusercontent.com/10363087/115101262-748c1d00-9f7d-11eb-8677-fe0ef1df8ff6.png">

<img width="864" alt="Screen Shot 2021-04-17 at 13 03 42" src="https://user-images.githubusercontent.com/10363087/115101265-78b83a80-9f7d-11eb-8fdd-4f19bac54ab4.png">

## Action Version & Workflow File

- Version v0.0.3

### Workflow File

Last one tried:
```yaml
name: Deploy

on:
  push:
    branches: [master]

jobs:
  build_and_deploy:
    name: Build & Deploy
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Build & Deploy to GitHub Pages
        uses: andrewmcodes/bridgetown-gh-pages-action@v0.0.3
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          repository: ${{ github.repository }}
          github_actor: ${{ github.actor }}
          default_branch: master # default_branch is not an expected input according to the GH actions log but is listed at https://github.com/andrewmcodes/bridgetown-gh-pages-action#inputs
          site_location: './'
          build_location: './output'
          deploy_branch: 'gh-pages'
          commit_message: 'new: release site'
```

**Comment by andrewmcodes** [comment]
hmm I’ll take a look when I get a chance.

**Comment by ianbayne** [comment]
Thanks, @andrewmcodes!

**Comment by ianbayne** [comment]
Has there been any work done on this?

**Comment by andrewmcodes** [comment]
@ianbayne Sorry this slipped off my radar. Is your site open source? I haven't tried this action with newer releases yet. It's on my todo list but I can debug your site if you can give me access. Also you should [join the Bridgetown Discord](https://discord.gg/4E6hktQGz4), lots of amazing folks who can help. 

Also just a thought, do you `.nojekyll` in your root? [GitHub Pages: Bypassing Jekyll Docs](https://github.blog/2009-12-29-bypassing-jekyll-on-github-pages/)

**Comment by ianbayne** [comment]
No problem at all, @andrewmcodes. Yes, here's the URL for it:

https://github.com/ianbayne/ianbayne.github.io

I appreciate the offer of help! 🙇 

> Also just a thought, do you .nojekyll in your root? GitHub Pages: Bypassing Jekyll Docs

I just tried adding this now but it doesn't seem to have helped.

**Comment by andrewmcodes** [comment]
@ianbayne I really didn't change much - I am pretty sure your site is not configured correctly in GitHub and that is what is causing the issue.

- [Clone of your site](https://github.com/andrewmcodes/andrewmcodes.github.io) - only 1 commit of changes that basically reverts some of the changes you added in the previous commit
- [Live Demo](https://andrewmcodes.github.io)
- Settings:

![Pages_Safari_20210630_170616](https://user-images.githubusercontent.com/18423853/124045882-83536f00-d9c5-11eb-8919-8ed9ca5997b8.png)

---

Take a look and _please_ let me know if it works so I can take your site down from my domain 😄 . 

One thing to note is that when create your initial commit to the repo, the action will then the `gh-pages` branch, so you actually have to deploy again to make it publish.

>Also just a thought, do you .nojekyll in your root? GitHub Pages: Bypassing Jekyll Docs

The action creates a `.nojekyll` file for you - I totally forgot that so it is unnecessary.

---

Now what?

1. Try the above changes (I really believe it's just the site config settings in GitHub - not the action. If my hunch is correct and you are able to get it working right from the UI, you'll likely want to revert [this change](https://github.com/ianbayne/ianbayne.github.io/commit/3aee720ade4276cd36e35fe1ae3f9be5d46327df)
2. If you are successful, let me know how we can update the docs to make it easier for others (whether here or Bridgetown's docs) to get this working in the future.
3. If you are unsuccessful let me know and we can debug, or possibly pair for an hour. Btw if you have any Bridgetown questions, I consider myself to be quite experienced and would love to help if you want to join the discord!
4. Whenever this issue is closed, I will be merging the changes for #5 & #6 and cutting a new major release (either `1.0.0` or `0.1.0`) 

One last thing to consider: this action, as ugly as it is, does work but I would consider it `prior art`. Why not move to a more modern platform that will make issues like this vanish. [Netlify](https://netlify.com) and [Vercel](https://vercel.com) are my favorites & I am happy to talk more about why. 

Again, sorry for my poor response times. Hop in discord for more immediacy or even ping me on [Twitter](https://twitter.com/andrewmcodes).

**Comment by ianbayne** [comment]
@andrewmcodes: Thank you very much! Indeed changing my GitHub Page's source from `master` to `gh-pages` was all it took. You're free to take my site down from your domain now 🤣 

I suppose I was confused because my site was previously a Jekyll site with its GitHub Page's source set to `master` and I both didn't have a good enough understanding of GitHub Pages nor did I look thoroughly enough through https://github.com/andrewmcodes/bridgetown-gh-pages-action/blob/main/action.yml to notice that the default deploy branch is set to `gh-pages`. So thanks again! 🙇 

My only feedback in regard to "[letting you] know how [to] update the docs to make it easier for others (whether here or Bridgetown's docs) to get this working in the future" would be to suggest that you note somewhere that the set GitHub Page's source and the deploy branch in the action need to be the same. I'm sure this is quite obvious for people more familiar with GitHub Pages, but for us newbs it could be quite helpful.

Thanks again—I really appreciate it! Shall close this issue now.
